lib/literal/array.rb

```diff
@@ -176,15 +176,22 @@ def pop(...)
 		@__value__.pop(...)
 	end
 
-	def push(value)
-		Literal.check(actual: value, expected: @__type__) do |c|
-			c.fill_receiver(receiver: self, method: "#push")
+	def push(*value)
+		case value
+		when ::Array
+			Literal::Array(@__type__).new(*value)
```

Comment:
This seemed like the best way of forcing the type check to happen on an array of values. But 🤷‍♂️